### PR TITLE
Remove EmailSubscription model

### PR DESCRIPTION
**Pull request by KludgeKML**
Brexit email subscriptions were held as EmailSubscription records in account-api which referenced Subscription records in email-alert-api. This is no longer the case, and in production there are only 5 records in the EmailSubscription model, all of which are for GDS developers and none of which refer to existing records in email-alert-api. Tidy up by removing this model and the supporting code.

https://trello.com/c/8x01txnU/20-brexit-checker-uses-email-alert-api-as-a-substitute-database

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

**Review by 1pretz1 — COMMENTED** [review]
Nice work Keith! Great to be removing some left over cruft. Few comments, two minor concerning the commits and one regarding the dropping of the table.

File: db/migrate/20230824083133_drop_email_subscriptions.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class DropEmailSubscriptions < ActiveRecord::Migration[7.0]
+  def down
+    drop_table :email_subscriptions
```
[the remark this answers is not in the document]
Comment (KludgeKML) [reply db/migrate/20230824083133_drop_email_subscriptions.rb:3]: Oh - yes, it had, but it was pointing in the wrong direction. D'oh! Fixed now.

**Comment by KludgeKML** [comment]
@1pretz1 Thanks for the comments good points - I think I've resolved them all now.

**Review by 1pretz1 — APPROVED** [review]
LGTM 🍏

**Pull request merged**